core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -335,7 +335,7 @@ private object RTS {
                   case IO.Tags.SyncEffect =>
                     val io = curIo.asInstanceOf[IO.SyncEffect[Any]]
 
-                    val value = io.effect()
+                    val value = io.effect(ExecutionContext.fromExecutor(rts.threadPool))
```

Comment:
You should put `ExecutionContext.fromExecutor(rts.threadPool)` into a field inside `rts` to avoid re-allocating memory for it every type there is a `SyncEffect`. Otherwise, looks great!

Reply:
Done 👍